crypto/core_namemap.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "internal/namemap.h"
+#include <openssl/lhash.h>
+#include <openssl/safestack.h>
+
+typedef struct {
+    int number;
+    const char *name;
+    char body[1];        /* Sized appropriately to contain the name */
+} NAMEMAP;
+
+DEFINE_LHASH_OF(NAMEMAP);
+DEFINE_STACK_OF(NAMEMAP)
```

Comment:
Sparse array instead here?
This kind of lookup is why it was added.

Reply:
Sure...

Reply:
Your discussion comments make me wonder, though, if you have the right idea about the code implemented here...

Reply:
I'm wrong :(